app/process/handler/crypto-transfer/crypto-transfer.go

```diff
@@ -51,30 +55,52 @@ func NewCryptoTransferHandler(
 		hederaNodeClient:   hederaNodeClient,
 		transactionRepo:    transactionRepository,
 		logger:             config.GetLoggerFor("Account Transfer Handler"),
+		ethereumClient:     ethereumClient,
 	}
 }
 
 // Recover mechanism
 func (cth *CryptoTransferHandler) Recover(q *queue.Queue) {
 	cth.logger.Info("[Recovery] Executing Recovery mechanism for CryptoTransfer Handler.")
-	cth.logger.Info("[Recovery] Database GET [PENDING] [SUBMITTED] transactions.")
+	cth.logger.Infof("[Recovery] Database GET [%s] [%s] [%s] transactions.", txRepo.StatusInitial, txRepo.StatusSignatureSubmitted, txRepo.StatusEthTxSubmitted)
 
-	transactions, err := cth.transactionRepo.GetPendingOrSubmittedTransactions()
+	transactions, err := cth.transactionRepo.GetIncompleteTransactions()
```

Comment:
I think that `crypto-transfer` handler is responsible for "recovering" only his logic, f.e if the status is initial, submit the TX again to the HCS topic or if the TX was submitted, check for its status.

I think that the `consensus-message` handler should be responsible for recovering its logic. Having said that, I think that it might be better if we do not return `StatusEthTxSubmitted` here and handle only `initial` and `signature_submitted`.

wdut @failfmi ?